shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -378,6 +444,36 @@ - (void)setupNotificationCenterObservers {
                object:nil];
 }
 
+- (void)setupKeyboardAnimationCurve {
+  _keyboardAnimationStops.reset([[NSMutableArray alloc] init]);
+
+  // Keyboard animation spring curve for iOS 16 and below
+  _keyboardSpringCurve.reset([[KeyboardSpringCurve alloc] initWithStiffness:1000
+                                                               dampingRatio:1
+                                                                       mass:3
+                                                            initialVelocity:0
+                                                           settlingDuration:0.5]);
+  double settlingDuration = [[self keyboardSpringCurve] settlingDuration];
+  double maxFramesPerSecond = [DisplayLinkManager displayRefreshRate];
+  double frameDuration = 1 / maxFramesPerSecond;
+
+  // Calculating a full seconds worth of animation stops for increased stop fidelity
+  for (int time = 1; time < maxFramesPerSecond; time++) {
+    double frameTime = time / maxFramesPerSecond;
+    NSNumber* keyboardAnimationStop =
```

Comment:
What if the animation time is longer than one second? (Although it is 0.5s now, but the algorithm should be with more flexibility ^_^

Reply:
Nice catch but unlikely to happen. It usually is 250ms or 500ms. 1sec would be too slow.

Reply:
I changed the loop bound to `roundf(maxFramesPerSecond * settlingDuration * 2)` for settlingDuration control (included the round to keep it an int evaluation). Think that makes sense